### DISCUSSION TITLE
calling functions in supabase cron job

### DISCUSSION THREAD

**Discussion by amanuela97** [supabase/supabase#5612] [Questions] [answered]
so i am trying to schedule a cron job in supbase sql editor where i call one of my function which returns a json from a weather api.
and then use that response to pass it as an argument to a second function. How can i access my functions inside the scheduler? and does it matter if the function was written with plv8 instead of plpgsql because i only get reference to functions i wrote in plpgsql inside the scope of the scheduler?

the code should do the follwoing:

``` select
  cron.schedule(
    'weather-every-minute', -- name of the cron job
    '* * * * *', -- every minute
    $$
     -- access and run function daily_task() here
     -- daily_task() returns json 
     -- take the json response and pass it as an argument to another function called send_email_message( response )
    $$
  );
```

**Comment by thorwebdev** [comment, marked as answer]
You can use [pg_net](https://supabase.com/docs/guides/database/extensions/pgnet) with [pg_cron](https://supabase.com/docs/guides/database/extensions/pgcron) to call a Supabase Edge Function:

```sql
select
  cron.schedule(
    'invoke-function-every-minute',
    '* * * * *', -- every minute
    $$
    select
      net.http_post(
          url:='https://project-ref.functions.supabase.co/function-name',
          headers:='{"Content-Type": "application/json", "Authorization": "Bearer YOUR_ANON_KEY"}'::jsonb,
          body:=concat('{"time": "', now(), '"}')::jsonb
      ) as request_id;
    $$
  );
```

In your Edge Function you can then call the weather API and send the email.

**Reply by samuel-messing** [reply]
@thorwebdev I have two environments for supabase, `dev`, and `prod`. Is there a way to dynamically populate `YOUR_ANON_KEY`?

**Reply by mellowcello77** [reply]
Great question - any answers yet? Can we access env variables from here?

**Reply by oscyp** [reply]
@samuel-messing @mellowcello77 you have to create a table named `config` or sth else with columns key, value and create a stored procedure which selects this `YOUR_ANON_KEY`. Example:

> CREATE OR REPLACE FUNCTION schedule_cron()
> RETURNS VOID AS $$
> DECLARE
>     anon_key_value TEXT;
> BEGIN
>     SELECT value INTO anon_key_value FROM config WHERE key = 'anon_key';
> 
>     PERFORM net.http_get(
>         url := concat('http://your-url.com/functions/v1/cron'),
>         headers := format('{"Content-Type": "application/json", "Authorization": "Bearer %s"}', anon_key_value)::jsonb
>     );
> END;
> $$ LANGUAGE plpgsql;
> 
> SELECT
> cron.schedule(
>     'cron',
>     '0 * * * *',
>     'SELECT schedule_cron()'
> );
